packages/zero-client/src/client/client-error.ts

```diff
@@ -0,0 +1,20 @@
+import type {OnlineManager} from './online-manager.ts';
+
+/**
+ * Client-side error thrown when the Zero client is in offline mode.
+ * Offline mode is a local state that disables mutators until connectivity
+ * is restored (or the mode is cancelled).
+ */
+export class OfflineError extends Error {
+  readonly name = 'OfflineError';
+  readonly code = 'OFFLINE';
+  constructor(message = 'Offline mode: Mutations are disabled while offline') {
+    super(message);
+  }
+}
+
+export function assertNotOffline(onlineManager: OnlineManager): void {
```

Comment:
The word `assert` in our codebase mean an invariant was violated - basically a bug in our code. Use a word like checkOffline or mustBeOnline or something.

Reply:
This can be called throwIfOffline